src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/AddDisableRuntimeMarshallingAttributeFixer.cs

```diff
@@ -0,0 +1,80 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.Interop.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class AddDisableRuntimeMarshallingAttributeFixer : CodeFixProvider
+    {
+        private const string EquivalenceKey = nameof(AddDisableRuntimeMarshallingAttributeFixer);
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(GeneratorDiagnostics.Ids.TypeNotSupported);
+
+        // TODO: Write a custom fix all provider
+        public override FixAllProvider? GetFixAllProvider() => null;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            List<Diagnostic> fixedDiagnostics = new(context.Diagnostics.Where(IsRequiresDiableRuntimeMarshallingDiagnostic));
+
+            if (fixedDiagnostics.Count > 0)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        "Add DisableRuntimeMarshallingAttribute to the assembly.",
+                        ct => AddDisableRuntimeMarshallingAttributeApplicationToProject(context.Document.Project, ct),
+                        EquivalenceKey),
+                    fixedDiagnostics);
+            }
+
+            return Task.CompletedTask;
+
+            static bool IsRequiresDiableRuntimeMarshallingDiagnostic(Diagnostic diagnostic)
+            {
+                return diagnostic.Properties.ContainsKey(GeneratorDiagnosticProperties.AddDisableRuntimeMarshallingAttribute);
+            }
+        }
+
+        private static async Task<Solution> AddDisableRuntimeMarshallingAttributeApplicationToProject(Project project, CancellationToken cancellationToken)
+        {
+            Document? assemblyInfo = project.Documents.FirstOrDefault(IsPropertiesAssemblyInfo);
+
+            if (assemblyInfo is null)
+            {
+                assemblyInfo = project.AddDocument("AssemblyInfo.cs", "", folders: new[] { "Properties" });
+            }
+
+            DocumentEditor editor = await DocumentEditor.CreateAsync(assemblyInfo, cancellationToken).ConfigureAwait(false);
+
+            var syntaxRoot = await assemblyInfo.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            editor.ReplaceNode(
+                syntaxRoot,
+                editor.Generator.AddAttributes(
+                    syntaxRoot,
+                    editor.Generator.Attribute(editor.Generator.DottedName(TypeNames.System_Runtime_CompilerServices_DisableRuntimeMarshallingAttribute))));
+
+            return editor.GetChangedDocument().Project.Solution;
+
+            static bool IsPropertiesAssemblyInfo(Document document)
+            {
+                return document.Name == "AssemblyInfo.cs"
+                    && document.Folders.Count == 1
```

Comment:
I specifically want to target this fix to put the attribute in a file at the `Properties/AssemblyInfo.cs` path from the project root. (This will also work for projects with file includes like `<Compile Include="$(CommonPathProperty)/Properties/AssemblyInfo.cs" Link="Properties/AssemblyInfo.cs" />` because of how Roslyn handles this.) With your replacement, any path under `Properties/` that has an `AssemblyInfo.cs` file could be selected, which I think can be a little unexpected.

Reply:
I'll add a comment documenting that we're exactly matching a particular path (and extract the string literals to constants).